packages/supi/src/install/index.ts

```diff
@@ -837,8 +837,16 @@ async function installInContext (
       })
     }
 
-    if (!opts.lockfileOnly) {
-      await Promise.all(projectsToLink.map(linkBinsOfImporter))
+    const binNames = (await Promise.all(projectsToLink.map(linkBinsOfImporter))).flat()
+    if (opts.global) {
+      const installPackages = projects
+        .map(project => project.wantedDependencies.reduce((acc: string[], wantedDependency) => [...acc, wantedDependency.alias], []))
+        .flat()
+      for (const name of installPackages) {
+        if (!binNames.includes(name)) {
+          logger.warn({ message: `The globally installed package ${name} has no bins.`, prefix: opts.lockfileDir })
+        }
+      }
```

Comment:
This solution will not work if the package's exported bin does not match the package name. It is better to move the check into line 840

Something like

```typescript
      await Promise.all(projectsToLink.map(async (project) =>{
         const linkedBins = awit linkBinsOfImporter(project
         if (!linkedBins.length) {
         ...
```

Reply:
still `linkBinsOfImporter` would return all the linked bins, including the ones installed before. We still need some sort of matching mechanism

Reply:
also I'm returning the `pkgName` not the exported bin name

Reply:
In that case, seems fine